packages/doenetml-worker-rust/lib-doenetml-core/src/core/math_via_wasm.rs

```diff
@@ -0,0 +1,327 @@
+/// - `parse_latex_into_math("g(x)", true, &["f", "g"])` is the function `g` evaluated at `x`.
+/// - `parse_latex_into_math("g(x)", true, &["f"])` is `g` times `x`.
+#[cfg(all(not(feature = "testing"), feature = "web"))]
+pub fn parse_latex_into_math<Latex: AsRef<str>, FnSymbol: AsRef<str>>(
+    latex: Latex,
+    split_symbols: bool,
+    function_symbols: &[FnSymbol],
+) -> Result<String, anyhow::Error> {
+    let js_function_symbols = JsValue::from(
+        function_symbols
+            .iter()
+            .map(|x| JsValue::from_str(x.as_ref()))
+            .collect::<Array>(),
+    );
+    let result: JsString = parseLatexIntoMath(
+        JsString::from(latex.as_ref()),
+        Boolean::from(split_symbols),
+        js_function_symbols,
+    )
+    .map_err(|e| anyhow!("{:?}", e))?;
+    Ok(result.into())
+}
+#[cfg(any(feature = "testing", not(feature = "web")))]
+pub fn parse_latex_into_math<Latex: AsRef<str>, FnSymbol: AsRef<str>>(
+    _latex: Latex,
+    _split_symbols: bool,
+    _function_symbols: &[FnSymbol],
+) -> Result<String, anyhow::Error> {
+    Err(anyhow!(
+        "parse_latex_into_math is only available when compiled with the `web` feature".to_string()
+    ))
+}
+
+/// Return a LaTeX string that corresponds to the mathematical expression `math_object`.
+/// The behavior is controlled by `params`.
+///
+/// Examples:
+/// ```
+/// let expr = parse_text_into_math("123 / 0.05", true, &["f"]).unwrap();
+///
+/// assert_eq!(
+///     math_to_latex(&expr, ToLatexParams::default()).unwrap(),
+///     r#"\frac{123}{0.05}"#
+/// );
+///
+/// let pad_three_decimals = ToLatexParams {
+///     pad_to_decimals: Some(3),
+///     ..Default::default()
+/// };
+/// assert_eq!(
+///     math_to_latex(&expr, pad_three_decimals).unwrap(),
+///     r#"\frac{123.000}{0.050}"#
+/// );
+///
+/// let pad_four_digits = ToLatexParams {
+///     pad_to_digits: Some(4),
+///     ..Default::default()
+/// };
+/// assert_eq!(
+///     math_to_latex(&expr, pad_four_digits).unwrap(),
+///     r#"\frac{123.0}{0.05000}"#
+/// );
+///
+/// let expr_with_blanks = parse_text_into_math("x + ()", true, &["f"]).unwrap();
+///
+/// assert_eq!(
+///     math_to_latex(&expr_with_blanks, ToLatexParams::default()).unwrap(),
+///     "x + \u{FF3F}"
+/// );
+///
+/// let hide_blanks = ToLatexParams {
+///     show_blanks: false,
+///     ..Default::default()
+/// };
+/// assert_eq!(
+///     math_to_latex(&expr_with_blanks, hide_blanks).unwrap(),
+///     "x + "
+/// );
+/// ````
+#[cfg(all(not(feature = "testing"), feature = "web"))]
+pub fn math_to_latex<MathObj: AsRef<str>>(
+    math_object: MathObj,
+    params: ToLatexParams,
+) -> Result<String, anyhow::Error> {
+    let result: JsString = toLatex(
+        JsString::from(math_object.as_ref()),
+        params.pad_to_decimals.map(Number::from),
+        params.pad_to_digits.map(Number::from),
+        Boolean::from(params.show_blanks),
```

Comment:
We could make this function more ergonomic to use if we grouped all the 2,3,4th arguments together as a `JsValue` (an object) and then defined `.into()` for `params`.
It culd look like `toLatex(JsString::from(math_object.as_ref()), params.into())`

Reply:
We could do that, but I'm reluctant to add it to these intermediate "throw away" functions that won't be used directly, given that it will add more complexity with more implementations depending on this conditional compilation nonsense.